### PR TITLE
Bazel: bugfix for join_paths and _remove_prefix

### PR DESCRIPTION
**Pull request by storypku**
---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/14553)

### CODE REVIEW

**Comment by jwnimmer-tri** [comment]
@drake-jenkins-bot ok to test

+@jwnimmer-tri for review.

File: tools/skylark/pathutils.bzl [hunk 1]
```diff
@@ -51,7 +51,7 @@ def __remove_prefix(path, prefix):
             # remaining components (len(prefix) - n - 1; the 1 is the current
             # prefix component) less one (since we need to keep at least one
             # component of the path).
-            k = len(path) - (len(prefix) - n - 2)
+            k = len(path) - (len(prefix) - n - 1)
```
Comment (storypku) [inline tools/skylark/pathutils.bzl:55]: For the `_remove_prefix` function, given `path=
"bazel-bin/cyber/mainboard/mainboard"`, and `prefix=
"bazel-bin/**/mainboard"`, we would expect that
`_remove_prefix(path, prefix)` returns "mainboard".
However, as the original calculation of `k` is wrong, `_remove_prefix(path, prefix)` in this case wouldn't work properly.

**Comment by jwnimmer-tri** [comment]
Thanks for this!  FYI I am suspicious of the overall pre-existing implementation (besides just this one fix), so I plan to take a few extra days to review the whole file, so I have a solid understanding before merging this.  Stay tuned.

**Comment by storypku** [comment]
> Thanks for this! FYI I am suspicious of the overall pre-existing implementation (besides just this one fix), so I plan to take a few extra days to review the whole file, so I have a solid understanding before merging this. Stay tuned.

Thanks for your instant response and waiting for further updates. In fact, I am from the https://github.com/ApolloAuto/apollo project and would like to bring/adapt the install mechanism you implemented into Apollo.

**Comment by jwnimmer-tri** [comment]
=> #14564

**Pull request closed without merging**